jerry-core/parser/js/opcodes-dumper.h

```diff
@@ -24,511 +24,28 @@
 #include "rcs-records.h"
 #include "scopes-tree.h"
 
-/**
- * Operand (descriptor of value or reference in context of parser)
- */
-class jsp_operand_t
+typedef enum type_t : uint8_t
+{
+  EMPTY, /**< empty operand */
+  STRING_LITERAL, /**< operand contains string literal value */
+  NUMBER_LITERAL, /**< operand contains number literal value */
+  REGEXP_LITERAL, /**< operand contains regexp literal value */
+  SIMPLE_VALUE, /**< operand contains a simple ecma value */
+  SMALLINT, /**< operand contains small integer value (less than 256) */
+  IDENTIFIER, /**< Identifier reference */
+  THIS_BINDING, /**< ThisBinding operand */
+  TMP, /**< operand contains byte-code register index */
+  IDX_CONST, /**< operand contains an integer constant that fits vm_idx_t */
+  UNKNOWN, /**< operand, representing unknown value that would be rewritten later */
+  UNINITIALIZED /**< uninitialized operand
+                 *
+                 *   Note:
+                 *      For use only in assertions to check that operands
+                 *      are initialized before actual usage */
+} jsp_type_t;
```

Comment:
Could you, please, add description comment to the enum and name it more specifically, like `jsp_opnd_type_t`?